posts/the-radon-nikodym-theorem-in-lean.md

```diff
@@ -0,0 +1,165 @@
+This function is known as the Radon-Nikodym derivative 
+(denoted by $\frac{\text{d}\mu}{\text{d}\nu}$) and is *essentially unique** whenever 
+it exists.
+
+In Lean, this is shown by `absolutely_continuous_iff_with_density_radon_nikodym_deriv_eq`
+and can be found in `measure_theory/decomposition/radon_nikodym`.
+While Radon-Nikodym is the main motivation, the proof of the theorem 
+itself is rather simple once we have the prerequisites, and thus, the project 
+spanned over multiple files, most of which can be found in `measure_theory/decomposition`.
+
+Of all the prerequisites, the most important are are the 
+Jordan decomposition theorem and the Lebesgue decomposition theorem.
+The Jordan decomposition theorem uniquely classifies signed measures and allows 
+us to express every signed measure as a difference between two (mutually singular) 
+positive measures. While the theorem itself follows from the signed Hahn decomposition, 
+defining the structure of Jordan decompositions in Lean was tricky. While 
+initially, the decomposition was defined as a proposition, thanks to the suggestions 
+from the maintainers, it was decided to define the decomposition as a structure. 
+```lean
+structure jordan_decomposition (α : Type*) [measurable_space α] :=
+(pos_part neg_part : measure α)
+[pos_part_finite : is_finite_measure pos_part]
+[neg_part_finite : is_finite_measure neg_part]
+(mutually_singular : pos_part ⊥ₘ neg_part)
+```
+This was important, as later on, it was discovered that we were able to 
+relax some conditions on the uniqueness property of the Lebesgue decomposition 
+by introducing scalar multiplication on Jordan decompositions[^2]. Furthermore, 
+as we would often like to transport between signed measures and Jordan 
+decompositions, it is easier to work with types rather than propositions since 
+this required a construction of an equivalence between the two types. 
+
+[^2]: Given a signed measure `s`, a measure `μ` and a real number `r`, one may show 
+`(r • s).singular_part μ = r • s.singular_part μ` without 
+requiring `have_lebesgue_decomposition s μ` by proving that the scalar product 
+with a real number and the equivalence between signed measures and Jordan 
+decompositions commute. 
+
+A similar situation was reached with the Lebesgue decomposition. The Lebesgue 
+decomposition states that given two $\sigma$-finite measures $\mu$ and $\nu$, 
+there exists a measurable function $f : \alpha \to \overline{\mathbb{R}}_{\ge 0}$ 
+and a finite measure $\xi$ such that $\xi$ is mutually singular with respect 
+to $\nu$ (denoted $\xi \perp \nu$) and $\mu = \xi + f\nu$.
+As with the Jordan decomposition theorem, it was not clear how to represent this 
+statement. In particular, it is important for us to be able to extract the aforementioned 
+$f$ from the decomposition as this is the Radon-Nikodym derivative. After different
+experimentations, the statement was in the end decided to be represented as 
```

Comment:
Hopefully this is helpful rather than nitpicking, but: this sentence is unnecessarily convoluted and indirect. How about "After some experiments, I decided to represent this condition using a class."?